connect/main.py

```diff
@@ -114,11 +114,13 @@ def connect():
 # 'line' command ("connect line")
 #
 @connect.command('line')
-@click.argument('linenum')
-def line(linenum):
-    """Connect to line via serial connection"""
-    # TODO: Stub
-    return
+@click.argument('target')
+@click.option('--devicename', '-d', is_flag=True, help="connect by name - if flag is set, interpret target as device name")
```

Comment:
I'm wondering whether we should use the "-d" flag to denote device hostname, or if we should create a separate subcommand, something like the following:

`connect line <line_number>`
`connect device <device_name>`

Thoughts?

Reply:
Either makes sense to me!
@yxieca Thoughts?

Reply:
I think the decision factor is more with the underline code in soncutil, right?

From what I see below. we are calling a same function with a flag, then it makes more sense to keep -d. However, if we have 2 separate command in underline console until, then it makes more sense to have 2 commands.

Reply:
I prefer two separate subcommands, as this fits more in the CLI structure used by most switches (many nested subcommands, very few flags), and this is what most NEs are familiar with. This would help them realize they have two completely different methods of connecting to a device. I think it is fine if the underlying consutil uses the '-d' flag, but I think the 'connect' command should have separate subcommands.